### PR TITLE
fix build on Qt 4, OSG 3.2.1, CMake 2.8.12

### PR DESCRIPTION
**Pull request by nim65s**
with:
- typo in the Qt4 OpenGL component
- avoid Qt::FindDirectChildrenOnl, which is only in Qt5
- remove an inclusion of osgQt/Version, not available in OSG 3.2.1
- add guards to avoid MOC parsing boost headers, ref
  https://stackoverflow.com/questions/15455178/qt4-cgal-parse-error-at-boost-join

This PR lacks proper testing for now, so please don't merge without testing :)

### CODE REVIEW

File: src/urdf-parser.cpp [hunk 1]
```diff
@@ -24,8 +24,6 @@
 #include <QStringList>
 #include <QtGlobal>
 
-#include <osgQt/Version>
```
Comment (jmirabel) [inline src/urdf-parser.cpp]: Do you have a problem with this ?
With OSG 3.4, this includes checks that current Qt version matches the one of osgQt.

Comment (jmirabel) [reply src/urdf-parser.cpp]: > remove an inclusion of osgQt/Version, not available in OSG 3.2.1

Alright, understood. Can you surround it with OSG_VERSION_GREATER_THAN ?

Comment (nim65s) [reply src/urdf-parser.cpp]: Will do

Comment (jmirabel) [reply src/urdf-parser.cpp]: Do you have any idea of how we can know the Qt version used by osgQt ?
ldd can provide the answer but it's very ugly...

Comment (jmirabel) [reply src/urdf-parser.cpp]: I think I know the answer.
`osgQt/Version` was added when they started to support Qt 4 and Qt 5. If not present, I guess it is safe to assume that osgQt uses Qt 4.

**Comment by nim65s** [comment]
already done it there: https://gepgitlab.laas.fr/gsaurel/gepetto-viewer/commit/4add5efc8ed4f7a611742e8222b066084744d01f

**Comment by nim65s** [comment]
Unit tests need the X server now ? https://gepgitlab.laas.fr/gsaurel/gepetto-viewer/-/jobs/14812

**Comment by jmirabel** [comment]
Hum, QApplication need it and SafeApplication inherits from it... It wouldn't be hard to do otherwise. For now, just consider this a not a mistake.

**Comment by jmirabel** [comment]
What's are the combination of dependencies tested on Gepgitlab ?

**Comment by nim65s** [comment]
Normally is only one combination:
- urdfdom-headers
- urdfdom
- openscenegraph
- openscenegraph-qt
- qt

But as PythonQt was required until you solved it in #89 , I added pythonQt from robotpkg (python 2.7 / Qt 4) in the images, but it was not compiled with `-DPythonQt_Wrap_QtAll=ON` nor `-DPythonQt_Extensions=ON`.

**Comment by nim65s** [comment]
This PR works on 16.04 & 18.04. On 14.04:
```
[…]/src/gui/dialog/dialogloadenvironment.cc:18:38: fatal error: ui_dialogloadenvironment.h: No such file or directory
```

**Comment by nim65s** [comment]
I think that `CMAKE_AUTOUIC` is not available on cmake 2.8.12

**Comment by nim65s** [comment]
That's it: https://stackoverflow.com/questions/44451872/cmake-2-8-12-2-autouic-not-working-bug-or-missing-feature

 So either we bump https://github.com/Gepetto/gepetto-viewer/blob/7ca4990a582832db5e19f91e21b3c748df125299/CMakeLists.txt#L20 to 3.0 and drop support for 14.04, or we can temporary add a hacky
```
if(${CMAKE_VERSION} VERSION_LESS 3.0)
    QT4_WRAP_UI(DIALOGS dialogloadenvironment.ui)
    add_custom_target(uic_cmake28_compat ALL DEPENDS ${DIALOGS})
endif()
```

**Comment by nim65s** [comment]
Let's not do that also for `AUTOMOC`, as it is in CMake 2.8.6, and bump CMake requirement instead…

**Comment by nim65s** [comment]
LGTM: https://gepgitlab.laas.fr/gsaurel/gepetto-viewer/pipelines/3574/builds

**Comment by jmirabel** [comment]
It isn't very hard to manually to the moc/ui but let's keep this for later.

**Comment by jmirabel** [comment]
> But as PythonQt was required until you solved it in #89 , I added pythonQt from robotpkg (python 2.7 / Qt 4) in the images, but it was not compiled with -DPythonQt_Wrap_QtAll=ON nor -DPythonQt_Extensions=ON.

I am more and more in favor of handling PythonQt as QCustomPlot. It would make life easier !

**Comment by nim65s** [comment]
I don't even know why we are using PythonQt… Is it only to get plots ?

**Comment by jmirabel** [comment]
Adding plugins is made a lot simpler thanks to PythonQt. It is used for plots and other stuffs.

**Pull request merged**